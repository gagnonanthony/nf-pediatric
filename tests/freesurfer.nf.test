nextflow_pipeline {

    options "-stub-run"

    name "Test nf-pediatric profile freesurfer"
    script "../main.nf"

    test("Freesurfer profile - freesurfer - should run successfully") {

        when {
            params {

                params.input = "$projectDir/tests/data/samplesheet_testtracking.csv"
                params.outdir = "test_results/"

                params.freesurfer = true
                params.fs_license = "https://www.dropbox.com/scl/fi/0s8lp6lydyd0rxawxb4jm/license.txt?rlkey=hz54oc0d4sor69avqphtrjvgn&st=9e0yij97&dl=0"

            }
        }

        then {
            assert workflow.success
            assert workflow.trace.tasks().size() == 3
        }

    }

    test("Freesurfer profile - fastsurfer - should run successfully") {

        when {
            params {

                params.input = "$projectDir/tests/data/samplesheet_testtracking.csv"
                params.outdir = "test_results/"

                params.freesurfer = true
                params.use_fastsurfer = true
                params.fs_license = "https://www.dropbox.com/scl/fi/0s8lp6lydyd0rxawxb4jm/license.txt?rlkey=hz54oc0d4sor69avqphtrjvgn&st=9e0yij97&dl=0"

            }
        }

        then {
            assert workflow.success
            assert workflow.trace.tasks().size() == 3
        }

    }

}
