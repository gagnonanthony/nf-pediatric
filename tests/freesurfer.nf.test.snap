{
    "Freesurfer profile - fastsurfer - should run successfully": {
        "content": [
            6,
            {
                "FASTSURFER": {
                    "fastsurfer": "2.3.3+0000000"
                },
                "Workflow": {
                    "nf/pediatric": "v1.0dev"
                }
            },
            [
                "cortical_area_lh.tsv",
                "cortical_area_rh.tsv",
                "cortical_thickness_lh.tsv",
                "cortical_thickness_rh.tsv",
                "cortical_volume_lh.tsv",
                "cortical_volume_rh.tsv",
                "multiqc",
                "multiqc/global_multiqc_data",
                "multiqc/global_multiqc_plots",
                "multiqc/global_multiqc_report.html",
                "pipeline_info",
                "pipeline_info/pipeline_software_mqc_versions.yml",
                "sub-test",
                "sub-test/anat",
                "sub-test/anat/sub-test_desc-preproc_T1w.nii.gz",
                "sub-test/anat/sub-test_fastsurfer",
                "sub-test/anat/sub-test_seg-BrainnetomeChild_desc-labels.json",
                "sub-test/anat/sub-test_seg-BrainnetomeChild_desc-labels.txt",
                "sub-test/anat/sub-test_seg-BrainnetomeChild_dseg.nii.gz",
                "sub-test/anat/sub-test_seg-BrainnetomeChild_dseg_dilated.nii.gz",
                "sub-test/anat/sub-test_seg-BrainnetomeChild_stat-lh_area.tsv",
                "sub-test/anat/sub-test_seg-BrainnetomeChild_stat-lh_thickness.tsv",
                "sub-test/anat/sub-test_seg-BrainnetomeChild_stat-lh_volume.tsv",
                "sub-test/anat/sub-test_seg-BrainnetomeChild_stat-rh_area.tsv",
                "sub-test/anat/sub-test_seg-BrainnetomeChild_stat-rh_thickness.tsv",
                "sub-test/anat/sub-test_seg-BrainnetomeChild_stat-rh_volume.tsv",
                "sub-test/anat/sub-test_seg-BrainnetomeChild_stat-subcortical.tsv",
                "sub-test/figures",
                "sub-test/figures/sub-test_labels_mqc.png",
                "sub-test/multiqc",
                "sub-test/multiqc/sub-test_multiqc_data",
                "sub-test/multiqc/sub-test_multiqc_plots",
                "sub-test/multiqc/sub-test_multiqc_report.html",
                "subcortical_volumes.tsv"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2025-01-02T23:01:18.627052"
    },
    "Freesurfer profile - freesurfer - should run successfully": {
        "content": [
            6,
            {
                "FASTSURFER": {
                    "fastsurfer": "2.3.3+0000000"
                },
                "Workflow": {
                    "nf/pediatric": "v1.0dev"
                }
            },
            [
                "cortical_area_lh.tsv",
                "cortical_area_rh.tsv",
                "cortical_thickness_lh.tsv",
                "cortical_thickness_rh.tsv",
                "cortical_volume_lh.tsv",
                "cortical_volume_rh.tsv",
                "multiqc",
                "multiqc/global_multiqc_data",
                "multiqc/global_multiqc_plots",
                "multiqc/global_multiqc_report.html",
                "pipeline_info",
                "pipeline_info/pipeline_software_mqc_versions.yml",
                "sub-test",
                "sub-test/anat",
                "sub-test/anat/sub-test_desc-preproc_T1w.nii.gz",
                "sub-test/anat/sub-test_fastsurfer",
                "sub-test/anat/sub-test_seg-BrainnetomeChild_desc-labels.json",
                "sub-test/anat/sub-test_seg-BrainnetomeChild_desc-labels.txt",
                "sub-test/anat/sub-test_seg-BrainnetomeChild_dseg.nii.gz",
                "sub-test/anat/sub-test_seg-BrainnetomeChild_dseg_dilated.nii.gz",
                "sub-test/anat/sub-test_seg-BrainnetomeChild_stat-lh_area.tsv",
                "sub-test/anat/sub-test_seg-BrainnetomeChild_stat-lh_thickness.tsv",
                "sub-test/anat/sub-test_seg-BrainnetomeChild_stat-lh_volume.tsv",
                "sub-test/anat/sub-test_seg-BrainnetomeChild_stat-rh_area.tsv",
                "sub-test/anat/sub-test_seg-BrainnetomeChild_stat-rh_thickness.tsv",
                "sub-test/anat/sub-test_seg-BrainnetomeChild_stat-rh_volume.tsv",
                "sub-test/anat/sub-test_seg-BrainnetomeChild_stat-subcortical.tsv",
                "sub-test/figures",
                "sub-test/figures/sub-test_labels_mqc.png",
                "sub-test/multiqc",
                "sub-test/multiqc/sub-test_multiqc_data",
                "sub-test/multiqc/sub-test_multiqc_plots",
                "sub-test/multiqc/sub-test_multiqc_report.html",
                "subcortical_volumes.tsv"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.1"
        },
        "timestamp": "2025-01-02T22:41:26.12873"
    }
}